src/NLog.StructuredLogging.Json.Tests/EndToEnd/ViaLayout/WithFailingLayout.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System.Linq;
+using NLog.Config;
+using NLog.Layouts;
+using NLog.StructuredLogging.Json.Tests.JsonWithProperties;
+using NLog.Targets;
+using NUnit.Framework;
+
+namespace NLog.StructuredLogging.Json.Tests.EndToEnd.ViaLayout
+{
+    public class WithFailingLayout
+    {
+        [Test]
+        public void WhenLayoutFails()
+        {
+            // arrange
+            const string loggerName = "failingLogger";
+            GivenLoggingIsConfiguredForTest(GivenFailingTarget(loggerName));
+            var logger = LogManager.GetLogger(loggerName);
+
+            // act
+            logger.ExtendedInfo("test message", new { prop1 = "value1", prop2 = 2 });
+
+            LogManager.Flush();
+
+            var output = LogManager.Configuration.LogMessage(loggerName).First();
+
+            Assert.That(output, Does.StartWith(
+                "{\"fail1\":\"Render failed: ApplicationException Test render fail\",\"flat1\":\"flat1\",\"TimeStamp\":\""));
+            Assert.That(output, Does.EndWith(
+                "\"prop1\":\"value1\",\"prop2\":\"2\"}"));
+        }
+
+        [Test]
+        public void WhenPropertyNamesAreDuplicated()
+        {
+            // arrange
+            const string loggerName = "duplicatingLogger";
+            GivenLoggingIsConfiguredForTest(GivenTargetWithDuplicates(loggerName));
+            var logger = LogManager.GetLogger(loggerName);
+
+            // act
+            logger.ExtendedInfo("test message", new { prop1 = "value1", prop2 = 2 });
+
+            LogManager.Flush();
+
+            var output = LogManager.Configuration.LogMessage(loggerName).First();
+
+            Assert.That(output, Does.StartWith(
+                "{\"duplicated\":\"value1\",\"attributes_duplicated\":\"value2\",\"TimeStamp\":\""));
+            Assert.That(output, Does.EndWith(
+                "\"prop1\":\"value1\",\"prop2\":\"2\"}"));
+        }
+
+        private void GivenLoggingIsConfiguredForTest(Target target)
```

Comment:
Should this and subsequent methods be moved to the `SimpleConfigurator`?

Don't mind either way